### ISSUE TITLE
Show which shops are likely to be outdated

### ISSUE THREAD

**Issue by fizwidget** [organicmaps/organicmaps#7036]
**Is your feature request related to a problem? Please describe.**
I’ve noticed that it’s quite common for shop data in OSM to be outdated. Shops seem particularly prone to this problem due to how often they change/move/close relative to other map features. 

For example, I find an ice cream shop on the map, but I arrive to find that it closed years ago and has been replaced by a sandwich shop 😩

**Describe the ideal solution**
Here’s one potential approach:
1. Organic Maps calculates a confidence score for each shop. This could simply be based on the date of its most recent edit, but could also take other things into account. 
2. Low-confidence shops receive special UI treatment. Perhaps their icons look faded or de-emphasied on the map, and a warning is displayed in the details card (e.g. “This information might be outdated”).
3. We could let the user to confirm whether the shop still exists. This would set “check_date” on the shop’s OSM entity, improving the data for everyone. 

This should help users avoid disappointment (like in the ice cream example above), and help improve OSM’a shop data. 

**Describe alternatives you have considered**
None. 

**Additional context**
I’ve noticed Google Maps does something similar when they have reason to believe a shop’s opening hours are outdated (a warning in yellow text is displayed above them IIRC).

**Comment by fizwidget** [comment]
Here’s the Google Maps warning I was thinking of:

![IMG_2857](https://github.com/organicmaps/organicmaps/assets/1172012/9642cebc-edaa-4fdf-821c-64c48c878995)

**Comment by Jean-BaptisteC** [comment]
Using last date edit is not a good idea, we have a lot of shops in OSM that stay many years in the same place without edit. To be sure shops exist maybe you can check with streetview website like Mapillary.
Another solution which is planned, is to update maps more regularly.

**Comment by eisams** [comment]
I was also going to request this enhancement

POI information in OSM is frequently outdated, hence OM should implement the check_date tag to show the age
The POI editor [EveryDoor](https://every-door.app) lets you confirm the presence of POIs and sets the check_date tag

This is not supported by any other navigation app currently, so someone needs to go first. That can hopefully start improving the state of POI information which is quite mediocre to be honest

Previously suggested in the discussion forum: https://github.com/orgs/organicmaps/discussions/2386

> Using last date edit is not a good idea, we have a lot of shops in OSM that stay many years in the same place without edit. To be sure shops exist maybe you can check with streetview website like Mapillary. Another solution which is planned, is to update maps more regularly.
This is not about that, and it is quite user hostile to suggest people check the presence in another service when OSM in theory has functionality for this

**Comment by RedAuburn** [comment]
`check_date` has 1.1m uses, but is only useful for verifying the shop's existence. We could warn the user that the existence was last checked at the tagged date (if older than 1.5 years) and allow the user to verify its existence if they are within 100m of it.

`check_date:opening_hours=*` has 112k uses, so we could do a similar warning and verification, but as part of the opening_hours fragment.

**Comment by fizwidget** [comment]
> Using last date edit is not a good idea, we have a lot of shops in OSM that stay many years in the same place without edit.

True, but I suspect a significant proportion of these shops have closed or moved, and nobody has got around to updating the OSM data yet. 

I suppose the prevalence of this problem would vary by region, but where I live it’s pretty common. 

> Another solution which is planned, is to update maps more regularly.

This would be nice, but I don’t think it’d help much with this particular problem. It’s not that Organic Maps is outdated relative to OSM - it’s that OSM is outdated relative to the real world.

**Comment by Zverik** [comment]
I'd be optimistic and propose a blue checkmark or something to mark the venue is "verified". By verification, any of these dates should be inside 6 months:

* `check_date` (420k objects as of now — only part of those are POI)
* `survey:date` (70k objects, same caveat)
* creation time (or modification provided it's version 1)

**Comment by biodranik** [comment]
Can someone propose to add an API to OSM like "Mark feature as actual", that will just update the last modified date of the feature/tags, without modifying anything else? That will greatly simplify the logic and make it easy for apps to validate features, without introducing any additional "check_date" tags.

**Comment by deevroman** [comment]
> Can someone propose to add an API to OSM like "Mark feature as actual", that will just update the last modified date of the feature/tags, without modifying anything else? That will greatly simplify the logic and make it easy for apps to validate features, without introducing any additional "check_date" tags.

0. The API allows you to increase the version of an object without making changes to coordinates/tags/etc.  But increasing the version does not mean the relevance of the object.  Vandalism, clarification of presets, alignment of points, correction of typos — all this updates the version, but does not confirm the existence of POI.  Therefore, if there is a check_date, then you need to rely on it.
1. If you want to implement metadata in the OSM data model, then the discussion is here: https://github.com/osmlab/osm-data-model/discussions/27
2. Let's be realistic: we can't wait for API changes that change anything in the data model.
3. Why is it difficult to update check_date?

**Comment by biodranik** [comment]
1. There is a use-case for all users: mark a POI as actual, "still there".
2. Another use-case is for OSMers/mappers only: highlight outdated POIs to come and recheck them. That may be better handled by specialized editors.

As I wrote before, even if a user sees some "last checked a month ago" or "last checked 5 years ago" labels, that doesn't mean that the POI is still there, or has actual info.

**Comment by deevroman** [comment]
> There is a use-case for all users: mark a POI as actual, "still there".

check_date = now-date

**Comment by RedAuburn** [comment]
Any changes to the OSM API would be so far off that any discussion is effectively theoretical.
The best option is to implement `check_date` tag support

**Comment by eisams** [comment]
> As I wrote before, even if a user sees some "last checked a month ago" or "last checked 5 years ago" labels, that doesn't mean that the POI is still there, or has actual info.

That's the point, if you know it was last checked a month ago you now it's 99% it's still there. If it's 5 years ago, maybe 50% depending on the business type

Check_date is what we have now to convey this, and it's used by StreetComplete and EveryDoor. Maybe in another lifetime we have something better

**Comment by pastk** [comment]
I wouldn't display POIs differently depending on the check_date.
But having e.g. "may not be up to date" somewhere in the place page could be useful. Accompanied by a "Confirm its there still" button :)

**Comment by biodranik** [comment]
So the idea is to show some kind of label for almost every selected object on the map that it's "not up to date", right?

**Comment by pastk** [comment]
We can/should tune the criteria so that its actually only a reasonable fraction of objects..

**Comment by eisams** [comment]
What are the steps to implement check_date support?

First add it to the map data, wait for the next map data release, then start playing with UI?

I did try compiling a new map dataset, but the iOS simulator just crashes if I introduce a new mwm file. (I have also never done any app development, so no clue how to really make the UI stuff)

**Comment by pastk** [comment]
> First add it to the map data, wait for the next map data release, then start playing with UI?

I doubt it makes sense to inflate release map data before its actually used.

> 
> I did try compiling a new map dataset, but the iOS simulator just crashes if I introduce a new mwm file. (I have also never done any app development, so no clue how to really make the UI stuff)

What's in logs when it crashes?
If its a debug build then it'll fail when checking mwm checksum, so you've got to either update the checksum in countries.txt or comment the check in storage.cpp.

Please create a draft PR with changed you've used to make new maps and we can continue there.

**Comment by eisams** [comment]
Thanks. The underlying issue was that I switched out the mwm file while the simulator was running, that just crashed the whole thing

When I started closing and launching the simulator it failed on the version check in storage.cpp

`  if (m_currentVersion == ptr->GetVersion() && size != GetCountryFile(countryId).GetRemoteSize())
    LOG(LERROR, ("Inconsistent MWM and version for", *ptr));`
    
The size of the mwm is indistinguishable, I first compiled one without just to test the generator pipeline and they're both 165.2 MB for Norway Central :)

**Comment by pastk** [comment]
> Why half a year? With OM update frequency that can be considered 8-9 months. That is clearly too low for libraries and churches, but a lot for shops in supermarkets and kiosks. So, a good average.

We can use different criteria for different POI groups also.

Also my eyebrows would probably rise if I tap on a train station and see a "Is it still there?" prompt :)

**Comment by biodranik** [comment]
There are already focused apps to explicitly motivate users to update POIs.

It would be great to focus on motivating people to contribute to OSM without annoying/unnecessary prompts. E.g. by making it easier to report a wrong place.
